### PR TITLE
add Auth Context to store request header(Authorization)

### PR DESCRIPTION
**Pull request by corgiboygsj**

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/rest/AbstractRestClient.java [hunk 1]
```diff
@@ -303,6 +313,29 @@ public void close() {
...
+
+    public void resetAuthContext() {
+        this.authContext.remove();
+    }
+
+    public String getAuthContext() {
+        return this.authContext.get();
+    }
+
+    private void buildAuthToRequest(Builder builder) {
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/rest/AbstractRestClient.java:331]: attachAuthToRequest

File: src/main/java/com/baidu/hugegraph/rest/AbstractRestClient.java [hunk 2]
```diff
@@ -303,6 +313,29 @@ public void close() {
...
+
+    public String getAuthContext() {
+        return this.authContext.get();
+    }
+
+    private void attachAuthToRequest(Builder builder) {
+        // Add auth header
+        String auth = getAuthContext();
+        if (StringUtils.isNotEmpty(auth)) {
+            builder.header(HttpHeaders.AUTHORIZATION, auth);
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/rest/AbstractRestClient.java:335]: can add test case to assert it's effective？

File: src/main/java/com/baidu/hugegraph/rest/AbstractRestClient.java [hunk 3]
```diff
@@ -303,6 +313,29 @@ public void close() {
...
+        this.authContext.remove();
+    }
+
+    public String getAuthContext() {
+        return this.authContext.get();
+    }
+
+    private void attachAuthToRequest(Builder builder) {
+        // Add auth header
+        String auth = getAuthContext();
```
Comment (Linary) [inline src/main/java/com/baidu/hugegraph/rest/AbstractRestClient.java:333]: this.getAuthContext()

Comment (Linary) [inline src/main/java/com/baidu/hugegraph/rest/AbstractRestClient.java:333]: Need check auth != null ?

**Review by Linary — CHANGES_REQUESTED** [review]

File: src/main/java/com/baidu/hugegraph/rest/AbstractRestClient.java (hunk 3 above)
Comment (corgiboygsj) [reply src/main/java/com/baidu/hugegraph/rest/AbstractRestClient.java:333]: not necessary

Comment (javeme) [reply src/main/java/com/baidu/hugegraph/rest/AbstractRestClient.java]: StringUtils.isNotEmpty can accept null?

Comment (corgiboygsj) [reply src/main/java/com/baidu/hugegraph/rest/AbstractRestClient.java]: > StringUtils.isNotEmpty can accept null?

yep

File: pom.xml [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
     <groupId>com.baidu.hugegraph</groupId>
     <artifactId>hugegraph-common</artifactId>
-    <version>1.8.8</version>
+    <version>1.8.9</version>
```
Comment (Linary) [inline pom.xml:9]: line 290 `<Implementation-Version>1.8.8.0</Implementation-Version> ` also need modify to 1.8.9.0

**Review by Linary — CHANGES_REQUESTED** [review]

**Comment by javeme** [comment]
to fix codecov

File: src/test/java/com/baidu/hugegraph/unit/rest/RestClientTest.java [hunk 5]
```diff
@@ -420,4 +420,59 @@ public void testAuthContext() {
...
+        Mockito.when(response.readEntity(String.class)).thenReturn("content");
+        Mockito.when(builder.delete()).thenReturn(response);
+        Mockito.when(builder.get()).thenReturn(response);
+
+        Whitebox.setInternalState(client, "target", target);
+
+        RestResult result;
+
+        result = client.delete("test", ImmutableMap.of());
+        Assert.assertEquals(200, result.status());
```
Comment (javeme) [inline src/test/java/com/baidu/hugegraph/unit/rest/RestClientTest.java:467]: assert the builder contains AUTHORIZATION header

File: src/test/java/com/baidu/hugegraph/unit/rest/RestClientTest.java [hunk 6]
```diff
@@ -404,4 +407,72 @@ public void testClose() {
...
+
+        result = client.delete("test", "id");
```
Comment (javeme) [inline src/test/java/com/baidu/hugegraph/unit/rest/RestClientTest.java:469]: test for each method:

1. setContext with different token
2. do get/post/put/delete
3. assert AUTHORIZATION header is expected

and also test it after resetAuthContext()

**Review by javeme — DISMISSED** [review]

**Review by Linary — APPROVED** [review]

**Review by javeme — APPROVED** [review]

**Pull request merged**